Lib/fontTools/fontBuilder.py

```diff
@@ -0,0 +1,548 @@
+
+        fontSet.topDictIndex.append(topDict)
+
+        self.font["CFF "] = newTable("CFF ")
+        self.font["CFF "].cff = fontSet
+
+    def setupGlyf(self, glyphs, calcGlyphBounds=True):
+        assert self.isTTF
+        self.font["loca"] = newTable("loca")
+        self.font["glyf"] = newTable("glyf")
+        self.font["glyf"].glyphs = glyphs
+        if hasattr(self.font, "glyphOrder"):
+            self.font["glyf"].glyphOrder = self.font.glyphOrder
+        if calcGlyphBounds:
+            self.calcGlyphBounds()
+
+    def setupFvar(self, axes, instances):
+        addFvar(self.font, axes, instances)
+
+    def setupGvar(self, variations):
+        gvar = self.font["gvar"] = newTable('gvar')
+        gvar.version = 1
+        gvar.reserved = 0
+        gvar.variations = variations
+
+    def calcGlyphBounds(self):
+        glyphTable = self.font["glyf"]
+        for glyph in glyphTable.glyphs.values():
+            glyph.recalcBounds(glyphTable)
+
+    def setupMetrics(self, tableTag, metrics):
+        assert tableTag in ("hmtx", "vmtx")
+        mtxTable = self.font[tableTag] = newTable(tableTag)
+        roundedMetrics = {}
+        for gn in metrics:
+            w, lsb = metrics[gn]
+            roundedMetrics[gn] = int(round(w)), int(round(lsb))
+        mtxTable.metrics = roundedMetrics
+
+    def setupHorizontalHeader(self, **values):
+        self._initTableWithValues("hhea", _hheaDefaults, values)
+
+    def setupVerticalHeader(self, **values):
+        self._initTableWithValues("vhea", _vheaDefaults, values)
+
+    def setupVerticalOrigins(self, verticalOrigins, defaultVerticalOrigin=None):
+        if defaultVerticalOrigin is None:
+            # find the most frequent vorg value
+            bag = {}
+            for gn in verticalOrigins:
+                vorg = verticalOrigins[gn]
+                if vorg not in bag:
+                    bag[vorg] = 1
+                else:
+                    bag[vorg] += 1
+            defaultVerticalOrigin = sorted(bag, key=lambda vorg: bag[vorg], reverse=True)[0]
+        self._initTableWithValues("VORG", {}, dict(VOriginRecords={}, defaultVertOriginY=defaultVerticalOrigin))
+        vorgTable = self.font["VORG"]
+        vorgTable.majorVersion = 1
+        vorgTable.minorVersion = 0
+        for gn in verticalOrigins:
+            vorgTable[gn] = verticalOrigins[gn]
+
+    def setupPost(self, keepGlyphNames=True, **values):
+        postTable = self._initTableWithValues("post", _postDefaults, values)
+        if self.isTTF and keepGlyphNames:
+            postTable.formatType = 2.0
+            postTable.extraNames = []
+            postTable.mapping = {}
+        else:
+            postTable.formatType = 3.0
+
+    def setupMaxp(self):
+        if self.isTTF:
+            defaults = _maxpDefaultsTTF
+        else:
+            defaults = _maxpDefaultsOTF
+        self._initTableWithValues("maxp", defaults, {})
+
+    def setupDSIG(self):
```

Comment:
hm, is this really needed? I've been reluctant in adding support for similar "dummy" DSIG in fontmake/ufo2ft, as I think this is a dirty hack, which is easy to do anyway as a post-process step if really necessary. We need to deprecate the table for good, instead of keeping it "alive" even though invalid.

Reply:
Well, it's not called by default, but it is definitely needed in some cases. I believe MS Word doesn't do OT features in a TTF unless there's a DSIG. Or something like that.

It's one of those ugly legacy things, and since FontBuilder is mostly about convenience, I think it's a good thing to have that functionality here.

Reply:
> I believe MS Word doesn't do OT features in a TTF unless there's a DSIG. Or something like that.

I'd be nice if somebody could verify this is still the case in current versions of MS Office and let us know. I tried several times in the OT mailing list to raise this issue about DSIG table spec being insufficient to implement, if not even incorrect (MS signtool utility seems to calculate checksums differently than what's written in the spec), but without success. But I gave up on DSIG long time ago.

Reply:
Yeah, DSIG is a mess, a pain, and should die. However, there remains a need to produce fonts that work in legacy Office versions.

How about I rename `setupDSIG()` to `setupDummyDSIG()` and add a doc string saying something about the legacy status of this?

Reply:
that would be better, I think, yeah